AuctionHouse/Overview.md

```diff
@@ -0,0 +1,101 @@
+* `canChangeSalePrice`: `canChangeSalePrice` is only intended to be used with AuctionHouses with `requireSignOff` set to `true`. This allows the Auction House to do complicated order matching to find the best price for the seller.
+* `sellerFeeBasisPoints`: the share of the sale the marketplace takes on all trades
+    
+## Listing and Bidding
+Once the marketplace is active, any trader can post an offer to "sell" an asset on the marketplace. When this happens, the Auction House does two things:
+
+1. Auction House creates a `Sell Order`: in other words, creates a PDA known as the `SellerTradeState` which represents the offer. Trade States are special PDAs that are very cheap in comparison to other PDAs / Accounts. This is because these PDAs only have 1 byte of data, which is the bump of the PDA. All the details of a listings such as list price, amount of tokens, mint account etc, are hashed into the seeds of the PDA and therefore the PDA acts as a "proof of existence" for that listing.
+
+![](https://i.imgur.com/ki27Ds8.png)
+
+2. Auction House also assigns the another PDA: `programAsSigner` PDA as the `Delegate`. Delegates are a feature of the Solana SPL-token program and are discussed in detail [here](https://spl.solana.com/token#authority-delegation). Delegation allows the Auction House Instance to pull assets out of a token account when a sale goes through at a later point.
+
+![](https://i.imgur.com/aIRl7Hb.png)
+
+Similar to the `Sell Order`, when a buyer places a bid, Auction House creates the `BuyerTradeState` PDA representing the bid. The bid amount (native or SPL tokens) then needs to be transferred manually to the `BuyerEscrowAccount` PDA, which holds this amount till the sale goes through.
+    
+## Executing a Sale
+As soon as the buyer and the seller have both posted there bids and offers respectively, the trade can be completed by calling the `executeSale` instruction, which is a permission-less crank: in other words, can be executed by anyone without any fee or reward.
+
+The `BuyerEscrowAccount` will transfer the tokens to the seller (minus Auction House fees). The seller's token account will send the asset to the buyer via a delegate transfer, completing a trade.
+![](https://i.imgur.com/gpAX63m.png)
+
+
+There are three trade scenarios in which the `executeSale` instruction may be called in different ways:
+
+* **Bid + Sale in same transaction**: Alice lists an asset, Bob matches the offer and the sale goes through immediately.
+* **Offer + Sale in same transaction**: Bob makes an offer of 5 SOL for an unlisted asset owned by Alice, Alice likes the offer and lists the asset for 5 SOL thus completing the sale immediately.
+* **Offer, Bid, Sale in three separate instructions**: Alice lists an asset for 5 SOL, Bob makes an offer for this asset for 3 SOL, Alice likes the offer and agrees to the sale.
```

Comment:
This is a good attempt at using the primitives explained in the previous section to explain additional concepts but the reader gets a bit lost.

This could do with:
- A clearer distinction between explanation and examples.
- A more explicit distinction between the parts you're trying to explain.
  - First, the "execute sale" primitive.
  - Then, the concept of "buying" by placing a bid and executing the sale in the same transaction for an existing listing.
  - Finally, the concept of "selling" by listing and executing the sale in the same transaction for an existing bid.
- A simple diagram for each scenario.